plugin-api/src/main/java/org/sonar/api/server/http/JakartaHttpRequest.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Sonar Plugin API
+ * Copyright (C) 2009-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.api.server.http;
+
+import jakarta.servlet.http.HttpServletRequest;
+import java.util.Enumeration;
+
+/**
+ * Implementation of {@link HttpRequest} based on a delegate of {@link HttpServletRequest} from the Jakarta Servlet API.
+ */
+public class JakartaHttpRequest implements HttpRequest {
```

Comment:
why is this class in the API?
Shouldn't it be up to the products implementing the API to implement the `HttpRequest` with the framework they want to use?

Reply:
Indeed, it makes sense to drop it here completely and let products implement it